### PR TITLE
Updating Diagrams, Doxygen Input Files & Documentation + Overridden Redaction Behavior

### PR DESCRIPTION
**Pull request by dmccoystephenson**
# Updating Diagrams, Doxygen Input Files, & Documentation
## Purpose
The purpose of these changes is to bring the data flow & class usage diagrams up to date, to include the general redaction header/implementation files with the input files for doxygen and to update the documentation in a few places.

## Additional Changes
- Fixed a line in `build_cdot.sh` referencing a deleted file.
- Deleted empty `development.md` file.
- Updated data flow diagram README.
- Removed some outdated information in `installation.md`
- Removed a TODO in `testing.md`
- Added links to `configuration.md`, `testing.md` & `troubleshooting.md` documents.
- Documented 'broker transport failure' error in `troubleshooting.md`
- Removed 'success' echo message from `build_cdot.sh` script since the script can fail due to authentication issues.

# Overridden Redaction Behavior
## Purpose
The general redaction behavior has been overridden in the case of required leaf members, bitstrings, and optional objects with required members.

### Required Leaf Members
In the case of required leaf members `angle`, `transmission`, `traction`, `abs`, `scs`, `brakeBoost` and `auxBrakes`, a specific value is given to the member rather than removing the member.

### Bitstrings
In the case of bitstrings, if any bits are specified to be removed, the entire bitstring object is removed. An exception to this is the `wheelBrakes` bitstring, which is required so it is handled a little differently.

Objects with solely 'True' or 'False' members will be identified as bitstrings.

### Optional Objects
In the case of optional objects `weatherProbe`, `status` and `speedProfile` which have required members, the entire object will be removed.

## Additional Changes
- Overridden redaction behavior has been documented in the General Redaction section of the PPM.
- `payload.data.partII.events.eventDisabledVehicle` has been changed to `payload.data.partII.value.events.eventDisabledVehicle` in the fieldsToRedact.txt file (provided for testing purposes)
- Some tests have been modified to account for members that are no longer removed when redacted.
- A new test BSM (and accompanying test) with no bitstrings has been introduced to the project.

## Testing
- Tests have been verified to all be passing.
- The PPM has been verified to build & boot locally w/ docker-compose with these changes.
- The PPM has been verified to deploy correctly in our dev & test cluster with these changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,7 +186,9 @@ The unit tests are also built when the solution is compiled. For information on
 The docker-compose-standalone.yml file is meant for local testing/troubleshooting.
 
 To utilize this, pass the -f flag to the docker-compose command as follows:
-> docker-compose -f docker-compose-confluent-cloud.yml up
+> docker-compose -f docker-compose-standalone.yml up
+
+Sometimes kafka will fail to start up properly. If this happens, spin down the containers and try again.
```
Comment (dan-du-car) [inline README.md:191]: Do we know why it failed to start up? Why restart the container resolve the issues?

File: docs/diagrams/data-flow/README.md [hunk 2]
```diff
@@ -6,12 +6,13 @@ The purpose of this diagram is to show how data flows through the Privacy Protec
 - The yellow ovals are kafka topics that the PPM consumes from and produces to.
 
 ## Data Flow Explanation
-1. The PPM pulls from the Asn1 Input Topic and passes the messge to the BSMHandler class.
+1. The PPM pulls from the Input Topic and passes the message to the BSMHandler class.
 1. Throughout the flow, if the BSMHandler encounters a parse error or missing data, the message will be dropped.
```
Comment (dan-du-car) [inline docs/diagrams/data-flow/README.md:10]: Can we copy the flow chart to the readme.md so we can see the flow chart and explanation at one page?

File: README.md (hunk 1 above)
Comment (dmccoystephenson) [reply README.md:191]: This is the relevant error:
> kafka_1      | [2023-03-06 18:58:30,514] ERROR Error while creating ephemeral at /brokers/ids/1001, node already exists and owner '72070068176224256' does not match current session '72070094776041472' (kafka.zk.KafkaZkClient$CheckedEphemeral)
kafka_1      | [2023-03-06 18:58:30,521] ERROR [KafkaServer id=1001] Fatal error during KafkaServer startup. Prepare to shutdown (kafka.server.KafkaServer)
kafka_1      | org.apache.zookeeper.KeeperException$NodeExistsException: KeeperErrorCode = NodeExists
kafka_1      |  at org.apache.zookeeper.KeeperException.create(KeeperException.java:126)
kafka_1      |  at kafka.zk.KafkaZkClient$CheckedEphemeral.getAfterNodeExists(KafkaZkClient.scala:1904)
kafka_1      |  at kafka.zk.KafkaZkClient$CheckedEphemeral.create(KafkaZkClient.scala:1842)
kafka_1      |  at kafka.zk.KafkaZkClient.checkedEphemeralCreate(KafkaZkClient.scala:1809)
kafka_1      |  at kafka.zk.KafkaZkClient.registerBroker(KafkaZkClient.scala:96)
kafka_1      |  at kafka.server.KafkaServer.startup(KafkaServer.scala:308)
kafka_1      |  at kafka.Kafka$.main(Kafka.scala:109)
kafka_1      |  at kafka.Kafka.main(Kafka.scala)

It looks like a broker already exists when attempting to create it. Spinning down the containers w/ `docker-compose down` should remove the existing broker.

File: docs/diagrams/data-flow/README.md (hunk 2 above)
Comment (dmccoystephenson) [reply docs/diagrams/data-flow/README.md:10]: Yup! Once https://github.com/CDOT-CV/jpo-cvdp/pull/22 is merged this will have been addressed.

Comment (dmccoystephenson) [reply docs/diagrams/data-flow/README.md:12]: The data flow chart has been added to the data flow README.

**Pull request merged**